### PR TITLE
RxJava support added to login system

### PR DESCRIPTION
**Pull request by dreadpool2**
*Network connection in login done by rx java

### CODE REVIEW

File: app/src/main/java/com/example/nikit/eventsapp/LoginActivity.java [hunk 1]
```diff
@@ -11,27 +11,24 @@
...
+import io.reactivex.schedulers.Schedulers;
 import butterknife.BindView;
 import butterknife.ButterKnife;
-import retrofit2.Call;
-import retrofit2.Callback;
-import retrofit2.Response;
 
 public class LoginActivity extends AppCompatActivity {
-    ApiInterface apiService;
-    @BindView(R.id.username_et)
```
Comment (nikit19) [inline app/src/main/java/com/example/nikit/eventsapp/LoginActivity.java]: Why ?

File: app/src/main/java/com/example/nikit/eventsapp/LoginActivity.java [hunk 2]
```diff
@@ -69,32 +66,30 @@ public void checkToken() {
...
+        compositeDisposable.add(ApiClient.getOpenEventAPI().login(login)
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(response -> {
+                        Log.d("Message","Token Saved / Logged In Succesfully !");
+                        String accessToken = response.getAccessToken();
+                        Toast.makeText(getApplicationContext(), "Success!" , Toast.LENGTH_LONG).show();
+                        TOKEN = accessToken;
+                        progressDialog.cancel();
+                        checkToken();
```
Comment (nikit19) [inline app/src/main/java/com/example/nikit/eventsapp/LoginActivity.java]: What if the user types an incorrect email and password ? Please add all the logic that you are removing back here. Plus pull the latest changes code has been updated

Comment (dreadpool2) [reply app/src/main/java/com/example/nikit/eventsapp/LoginActivity.java]: Sure

**Comment by dreadpool2** [comment]
@nikit19 No issues here I guess?

File: app/src/main/java/com/example/nikit/eventsapp/MainActivity.java [hunk 3]
```diff
@@ -10,12 +10,13 @@
...
 import android.view.MenuItem;
 
-import com.example.nikit.eventsapp.utils.SharedPreferencesUtil;
+import butterknife.BindView;
+import butterknife.ButterKnife;
 
 public class MainActivity extends AppCompatActivity {
-
+    @BindView(R.id.navigation)
+    BottomNavigationView navigation;
```
Comment (nikit19) [inline app/src/main/java/com/example/nikit/eventsapp/MainActivity.java]: Is this needed ?

File: app/src/main/java/com/example/nikit/eventsapp/LoginActivity.java [hunk 4]
```diff
@@ -4,36 +4,36 @@
...
 public class LoginActivity extends AppCompatActivity {
-    ApiInterface apiService;
     @BindView(R.id.username_et)
     protected EditText usernameET;
     @BindView(R.id.password_et)
     protected EditText passwordET;
     @BindView(R.id.login_btn)
-    protected Button loginBtn;
+    protected AppCompatButton loginBtn;
+
```
Comment (nikit19) [inline app/src/main/java/com/example/nikit/eventsapp/LoginActivity.java]: Why appcompat?

File: app/src/main/java/com/example/nikit/eventsapp/LoginActivity.java [hunk 5]
```diff
@@ -4,36 +4,36 @@
...
     ProgressDialog progressDialog;
     public static String TOKEN=null;
     private SharedPreferencesUtil sharedPreferencesUtil;
+    private CompositeDisposable compositeDisposable = new CompositeDisposable();
+
```
Comment (nikit19) [inline app/src/main/java/com/example/nikit/eventsapp/LoginActivity.java]: Please avoid unnecessary blank lines

File: app/src/main/java/com/example/nikit/eventsapp/LoginActivity.java [hunk 6]
```diff
@@ -46,60 +46,60 @@ protected void onCreate(Bundle savedInstanceState) {
...
 
-        loginBtn.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
-                //sendPost(ei.getText().toString(),e2.getText().toString());
-                sendPost("hey@hey.hey", "heyheyhey");
-                progressDialog.show();
+        loginBtn.setOnClickListener(v -> {
+            //sendPost(ei.getText().toString(),e2.getText().toString());
+            LoginActivity.this.sendPost("hey@hey.hey", "heyheyhey");
```
Comment (nikit19) [inline app/src/main/java/com/example/nikit/eventsapp/LoginActivity.java]: You can remove this line and take the value from the user

File: app/src/main/java/com/example/nikit/eventsapp/MainActivity.java (hunk 3 above)
Comment (dreadpool2) [reply app/src/main/java/com/example/nikit/eventsapp/MainActivity.java]: @nikit19 We are referencing this view in the on create method. We are setting up the listener

File: app/src/main/java/com/example/nikit/eventsapp/LoginActivity.java (hunk 4 above)
Comment (dreadpool2) [reply app/src/main/java/com/example/nikit/eventsapp/LoginActivity.java]: @nikit19 In the xml file we do use an appcompat button. Won't matter though if I would change this to a Button. Just to be specific

File: app/src/main/java/com/example/nikit/eventsapp/MainActivity.java (hunk 3 above)
Comment (nikit19) [reply app/src/main/java/com/example/nikit/eventsapp/MainActivity.java]: Okay

**Pull request merged**